.github/workflows/update_version_numbers.yml

```diff
@@ -0,0 +1,57 @@
+name: Update OpenMS version numbers
+
+# Trigger this action manually
+
+variables:
+  package_version_major: 3
+  package_version_minor: 0
+  package_version_patch: 0
+  package_version: "$version_major.$version_minor.$version_patch"
+
+
+jobs:
+  updating_versions: # checking if the package versions in the files are not updated
+    - runs-on: macos-latest
+    - run: |
+      # update main cmakelist
+      sed -i '' "s#.*set(OPENMS_PACKAGE_VERSION_MAJOR.*#set(OPENMS_PACKAGE_VERSION_MAJOR \"$package_version_major\")#" ./CMakeLists.txt
+      sed -i '' "s#.*set(OPENMS_PACKAGE_VERSION_MINOR.*#set(OPENMS_PACKAGE_VERSION_MINOR \"$package_version_minor\")#" ./CMakeLists.txt
+      sed -i '' "s#.*set(OPENMS_PACKAGE_VERSION_PATCH.*#set(OPENMS_PACKAGE_VERSION_PATCH \"$package_version_patch\")#" ./CMakeLists.txt
+
+      # update version info test
+      sed -i '' "s#.*detail.version_major =.*#\tdetail.version_major = $package_version_major;#" ./src/tests/class_tests/openms/source/VersionInfo_test.cpp
+      sed -i '' "s#.*detail.version_minor =.*#\tdetail.version_major = $package_version_minor;#" ./src/tests/class_tests/openms/source/VersionInfo_test.cpp
+      sed -i '' "s#.*detail.version_patch =.*#\tdetail.version_major = $package_version_patch;#" ./src/tests/class_tests/openms/source/VersionInfo_test.cpp
+
+      # update test write ini out:
+      sed -i '' "s#<ITEM name=\"version\" value=\".*\" type=\"string\"#<ITEM name=\"version\" value=\"$package_version\" type=\"string\"#g" ./src/tests/topp/WRITE_INI_OUT.ini
+
+      # update INIs in tests topp:
+      find ./src/tests/topp/ -type f -name '*.ini' -exec grep "<ITEM name=\"version\" value=\".*\" type=\"string\"" {} \; -exec sed -i '' "s#<ITEM name=\"version\" value=\".*\" type=\"string\"#<ITEM name=\"version\" value=\"$package_version\" type=\"string\"#g" {} \;
+
+
+  commit_updated_files: # if any file is not updated, update files and push a commit
+    needs: updating_versions
+
+    steps:
+# Get updated files
+      - name: Get changed files
```

Comment:
I think simply doing a git diff --exit-code and checking the exit code will do.
This action is more for pull requests where you don't know what is the base branch etc. and it is very very unstable.

Reply:
That's a great idea! Thanks